vdoctl/cmd/drivers.go

```diff
@@ -0,0 +1,351 @@
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/vdoctl/pkg/utils"
+	"strings"
+
+	"github.com/thanhpk/randstr"
+	"github.com/vmware-tanzu/vsphere-kubernetes-drivers-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"os"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type credentials struct {
+	errorMsg            string
+	username            string
+	password            string
+	vcIp                string
+	insecure            bool
+	datacenters         []string
+	vSANDataStoresUrl   []string
+	vsphereCloudConfig  string
+	topology            v1alpha1.TopologyInfo
+	netPermissions      []v1alpha1.NetPermission
+	vsphereCloudConfigs []string
+	thumbprint          string
+}
+
+const (
+	KubeSystemNamespace = "kube-system"
+	vdoConfigName       = "vdo-config"
+	secretType          = "kubernetes.io/basic-auth"
+	ClusterDistribution = "OpenShift"
+)
+
+// driversCmd represents the drivers command
+var driversCmd = &cobra.Command{
+	Use:   "drivers",
+	Short: "Command to configure VDO",
+	Long:  `This command helps to specify the details required to configure CloudProvider and Storage Provider drivers.`,
+
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+
+		cpi := credentials{}
+		csi := credentials{}
+
+		thumbprintMap := make(map[string]string)
+
+		var isCPIMultiVC bool
+		var vsphereCloudConfigList, vcIPList []string
+
+		labels := credentials{
+			errorMsg: "unable to get VC creds",
+			username: "Username",
+			password: "Password",
+			vcIp:     "VC_IP",
+			topology: v1alpha1.TopologyInfo{
+				Zone:   "Zones",
+				Region: "Regions",
+			},
+			thumbprint: "SSL Thumbprint",
+		}
+
+		isCPIRequired := utils.PromptGetInput("Do you want to configure Cloud Provider? (Y/N)", errors.New("invalid input"), utils.IsString)
+
+		if isCPIRequired == "Y" || isCPIRequired == "y" {
+			fetchVCIP(&cpi, labels, "Cloud Provider")
+			vcIPList = append(vcIPList, cpi.vcIp)
+			if !cpi.insecure {
+				thumbprintMap[cpi.vcIp] = cpi.thumbprint
+			}
+
+		multivcloop:
+			for {
+
+				fetchCredentials(&cpi, labels, "Cloud Provider")
+
+				secret, err := createSecret(K8sClient, ctx, cpi, "cpi")
```

Comment:
Here before we create the secret can we validate the credentials with the given details.
Alert the user if the credentials entered is incorrect.
you can make this a todo enhancement on this flow. It is good to give early feedback in case of errors to users rather than later